idna/src/uts46.rs

```diff
@@ -453,7 +453,7 @@ impl Idna {
             return Errors::default();
         }
         let mut errors = processing(domain, self.config, &mut self.normalized, out);
-        self.output = std::mem::replace(out, String::with_capacity(out.len()));
+        self.output = core::mem::replace(out, String::with_capacity(out.len()));
```

Comment:
FYI, this fixes a bug in the current `idna` no_std support. Why is it not caught in CI, and should I open an extra PR for it?

Reply:
I recently dabbled with `no_std` stuff as well and I think the sure-fire way to test `no_std` compatibility is to:
- target a platform that does not have `std` ([certain ARM platforms](https://github.com/rust-random/rand/blob/master/.github/workflows/test.yml#L163)) OR
- have a really simple `no_std`  binary crate which uses the libraries

Reply:
Yes I'm trying it with `cargo +nightly build -Zbuild-std=core,alloc --target aarch64-unknown-none -v --release --no-default-features`

Reply:
Could you split this off into a separate PR?